velox/functions/prestosql/types/IPPrefixType.cpp

```diff
@@ -0,0 +1,329 @@
+
+namespace {
+
+class IPPrefixCastOperator : public exec::CastOperator {
+ public:
+  bool isSupportedFromType(const TypePtr& other) const override {
+    switch (other->kind()) {
+      case TypeKind::VARCHAR:
+        return true;
+      case TypeKind::HUGEINT:
+        if (isIPAddressType(other)) {
+          return true;
+        }
+      default:
+        return false;
+    }
+  }
+
+  bool isSupportedToType(const TypePtr& other) const override {
+    switch (other->kind()) {
+      case TypeKind::VARCHAR:
+        return true;
+      case TypeKind::HUGEINT:
+        if (isIPAddressType(other)) {
+          return true;
+        }
+      default:
+        return false;
+    }
+  }
+
+  void castTo(
+      const BaseVector& input,
+      exec::EvalCtx& context,
+      const SelectivityVector& rows,
+      const TypePtr& resultType,
+      VectorPtr& result) const override {
+    context.ensureWritable(rows, resultType, result);
+
+    if (input.typeKind() == TypeKind::VARCHAR) {
+      castFromString(input, context, rows, *result);
+    } else if (isIPAddressType(input.type())) {
+      castFromIPAddress(input, context, rows, *result);
+    } else {
+      VELOX_UNSUPPORTED(
+          "Cast from {} to IPPrefix not yet supported",
+          input.type()->toString());
+    }
+  }
+
+  void castFrom(
+      const BaseVector& input,
+      exec::EvalCtx& context,
+      const SelectivityVector& rows,
+      const TypePtr& resultType,
+      VectorPtr& result) const override {
+    context.ensureWritable(rows, resultType, result);
+
+    if (resultType->kind() == TypeKind::VARCHAR) {
+      castToString(input, context, rows, *result);
+    } else if (isIPAddressType(resultType)) {
+      castToIPAddress(input, context, rows, *result);
+    } else {
+      VELOX_UNSUPPORTED(
+          "Cast from IPPrefix to {} not yet supported", resultType->toString());
+    }
+  }
+
+ private:
+  static void castToString(
+      const BaseVector& input,
+      exec::EvalCtx& context,
+      const SelectivityVector& rows,
+      BaseVector& result) {
+    auto* flatResult = result.as<FlatVector<StringView>>();
+    const auto* ipaddresses = input.as<SimpleVector<StringView>>();
+
+    context.applyToSelectedNoThrow(rows, [&](auto row) {
+      const auto intAddr = ipaddresses->valueAt(row);
+      folly::ByteArray16 addrBytes;
+      std::string s;
+
+      memcpy(&addrBytes, intAddr.data(), kIPAddressBytes);
+      folly::IPAddressV6 v6Addr(addrBytes);
+
+      exec::StringWriter<false> result(flatResult, row);
+      if (v6Addr.isIPv4Mapped()) {
+        result.append(
+            v6Addr.createIPv4().str() + "/" +
+            std::to_string((uint8_t)intAddr.data()[kIPAddressBytes]));
+      } else {
+        result.append(
+            v6Addr.str() + "/" +
+            std::to_string((uint8_t)intAddr.data()[kIPAddressBytes]));
+      }
+      result.finalize();
+    });
+  }
+
+  static auto splitIpSlashCidr(folly::StringPiece ipSlashCidr) {
```

Comment:
I can switch to string view, but the function will get a little messy in that it will be switched to a find + creation of a vector<string_view> + creation of these string_views. This is used for error reporting and is taken straight from

https://github.com/facebook/folly/blob/556e17ac7b1f6cda68de789d5b06e79b2d3b990f/folly/IPAddress.cpp#L75

Please let me know. Thanks!

Reply:
Update.
Maybe misinterpreted the comment. Changed function argument to std::string_view. Function itself is still using StringPiece.

Reply:
Reverted back to StringPiece, however switched to const &. string_view caused string output issue.
```
Expected error message to contain 'Invalid IP address '2001:db8::1::1'', but received 'Invalid IP address 'ck\0\0\0\0\0\0_����''.
```